src/System/Random/Stateful.hs

```diff
@@ -642,6 +641,46 @@ runSTGen_ g action = fst $ runSTGen g action
 -- >   thawGen = MWC.restore
 -- >   freezeGen = MWC.save
 --
+-- === @FrozenGen@
+--
+-- `FrozenGen` gives us ability to use any stateful pseudo-random number generator in its
+-- immutable form, if one exists that is. This concept is commonly known as a seed, which
+-- allows us to save and restore the actual mutable state of a pseudo-random number
+-- generator. The biggest benefit that can be drawn from a polymorphic access to a
+-- stateful pseudo-random number generator in a frozen form is the ability to serialize,
+-- deserialize and possibly even use the stateful generator in a pure setting without
+-- knowing the actual type of a generator ahead of time. For example we can write a
+-- function that accepts a frozen state of some pseudo-random number generator and
+-- produces a short list with random even integers.
+--
+-- >>> import Data.Int (Int8)
+-- >>> :{
+-- myCustomRandomList :: FrozenGen f m => f -> m [Int8]
+-- myCustomRandomList f =
```

Comment:
`myCustomRandomList` -> `randomEvens` perhaps? Or is the name cumbersome on purpose so it stands out in the example code?

Reply:
Yes it was on purpose. Point is not what function produces, but how it is being used.